### PR TITLE
Add de-query-tool

### PR DESCRIPTION
**Pull request by BrunoCoimbra**
This addresses issue #111 requested by @StevenCTimm.

The new **de-query-tool** fetches all generations of a product from one or more recent task managers.
I made some changes to the PostgreSQL datasoruce to make it possible to query data products by name.

### CODE REVIEW

**Comment by jcpunk** [comment]
CodeCov is reporting a lot of missed lines on this (see the comments on the "files changed" tab)

**Comment by mambelli** [comment]
In  src/decisionengine/framework/tests/test_query_tool_server.py there are 2 functions w/ the same name. Then more lines should be covered by tests. On the exception you can add the pragma to skip the coverage test.

File: src/decisionengine/framework/engine/DecisionEngine.py [hunk 1]
```diff
@@ -448,6 +451,59 @@ def reaper_status(self):
...
+                                                         sequence_id=tm['sequence_id'])
+                        products = data_block.get_dataproducts(product)
+                        for p in products:
+                            df = p["value"]
+                            if df.shape[0] > 0:
+                                df["channel"] = [tm["name"]] * df.shape[0]
+                                df["taskmanager_id"] = [p["taskmanager_id"]] * df.shape[0]
+                                df["generation_id"] = [p["generation_id"]] * df.shape[0]
+                                result = result.append(df)
+                    except Exception as e:
```
Comment (mambelli) [inline src/decisionengine/framework/engine/DecisionEngine.py:491]: Can add pragma not to count in coverage

File: src/decisionengine/framework/engine/de_query_tool.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    de_socket = xmlrpc.client.ServerProxy(url, allow_none=True)
+    try:
+        return execute_command_from_args(args, de_socket)
+    except OSError as e:
+        msg = f"An error occurred while trying to access a DE server at '{url}'\n" + \
+            "Please ensure that the host and port names correspond to a running DE instance."
+        if args.verbose:
+            msg += f'\n{e}'
+        return msg
+    except Exception as e:
```
Comment (mambelli) [inline src/decisionengine/framework/engine/de_query_tool.py:59]: Pragma for coverage should go to all generic exceptions

**Review by mambelli — CHANGES_REQUESTED** [review]
For this time can be OK but whenever changing a docstring it should be changed to our standard Google format. 
There are 2 comments to add the pragmas to skip for coverage counting.

**Review by mambelli — APPROVED** [review]

**Pull request merged**